### PR TITLE
Move bernoulli further into ATen

### PR DESCRIPTION
**Pull request by cpuhrsch**
This addresses some of the points mentioned [in this issue](https://github.com/pytorch/pytorch/issues/6940) and should generally improve perf of some of those operations (it avoids a few unnecessary expands).

### CODE REVIEW

File: aten/src/ATen/native/Distributions.cpp [hunk 1]
```diff
@@ -108,14 +108,48 @@ int64_t sample_poisson(double lambda, THGenerator* generator) {
...
+Tensor bernoulli(const Tensor& self, const Tensor& p, Generator* gen) {
+  Tensor result = self.type().tensor();
+  result.resize_(self.sizes());
+  return native::bernoulli_(result, p, gen);
+}
+
+Tensor bernoulli(const Tensor& self, double p, Generator* gen) {
+  Tensor result = self.type().tensor();
+  result.resize_(self.sizes());
+  return native::bernoulli_(result, (double)p, gen);
```
Comment (apaszke) [inline aten/src/ATen/native/Distributions.cpp]: nit: it is already a double

File: aten/src/ATen/native/Distributions.cpp [hunk 2]
```diff
@@ -108,14 +108,48 @@ int64_t sample_poisson(double lambda, THGenerator* generator) {
...
+}
+
+Tensor bernoulli(const Tensor& self, Generator* gen) {
+  return native::bernoulli(self, (double)0.5, gen);
```
Comment (apaszke) [inline aten/src/ATen/native/Distributions.cpp]: same here

**Review by apaszke — APPROVED** [review]
Will conflict with #7171 😕

**Comment by MlWoo** [comment]
@apaszke It will be OK to solve the confiict for me.

**Comment by ezyang** [comment]
```
19:26:38 ======================================================================
19:26:38 ERROR: test_bernoulli (test_torch.TestTorch)
19:26:38 ----------------------------------------------------------------------
19:26:38 Traceback (most recent call last):
19:26:38   File "test_torch.py", line 6062, in test_bernoulli
19:26:38     t.bernoulli_(p)
19:26:38 RuntimeError: expected scalar type % s but found %sByteDouble
19:26:38 
19:26:38 ----------------------------------------------------------------------
```

NB: the error reporting format for `AT_ERROR` and co has changed, don't use printf strings.

**Comment by cpuhrsch** [comment]
I restarted the failed build which is now running here: https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-linux-trusty-py3.6-gcc7.2-trigger/7061/

It's a known issue that the github web ui doesn't pick up on it.

**Pull request merged**